video/out/drm_atomic.c

```diff
@@ -197,51 +196,53 @@ struct drm_atomic_context *drm_atomic_create_context(struct mp_log *log, int fd,
+        drmModeFreePlane(drmplane);
+        drmplane = NULL;
 
-        drmplane = drmModeGetPlane (ctx->fd, plane_res->planes[j]);
+        drmplane = drmModeGetPlane(ctx->fd, plane_res->planes[j]);
         if (drmplane->possible_crtcs & (1 << crtc_index)) {
             plane = drm_object_create(log, ctx->fd, drmplane->plane_id,
                                       DRM_MODE_OBJECT_PLANE);
 
-            if (plane) {
-                if (drm_object_get_property(plane, "TYPE", &value) == -EINVAL) {
-                    mp_err(log, "Unable to retrieve type property from plane %d\n", j);
-                    goto fail;
-                } else {
-                    if (value == DRM_PLANE_TYPE_CURSOR) // Skip cursor planes
-                        continue;
+            if (!plane) {
+                mp_err(log, "Failed to create Plane object from plane ID %d\n",
+                       drmplane->plane_id);
+                goto fail;
+            }
 
-                    layercount++;
+            if (drm_object_get_property(plane, "TYPE", &value) == -EINVAL) {
+                mp_err(log, "Unable to retrieve type property from plane %d\n", j);
+                goto fail;
+            }
 
-                    if ((!primary_id) && (value == DRM_PLANE_TYPE_PRIMARY))
-                        primary_id = drmplane->plane_id;
+            if (value != DRM_PLANE_TYPE_CURSOR) { // Skip cursor planes
+                layercount++;
 
-                    if ((!overlay_id) && (value == DRM_PLANE_TYPE_OVERLAY))
-                        overlay_id = drmplane->plane_id;
+                if ((!primary_id) && (value == DRM_PLANE_TYPE_PRIMARY))
+                    primary_id = drmplane->plane_id;
 
-                    if (layercount == osd_plane_id) {
-                        ctx->osd_plane = plane;
-                        continue;
-                    }
+                if ((!overlay_id) && (value == DRM_PLANE_TYPE_OVERLAY))
+                    overlay_id = drmplane->plane_id;
 
-                    if (layercount == video_plane_id) {
-                        ctx->video_plane = plane;
-                        continue;
-                    }
+                if (layercount == osd_plane_id) {
+                    ctx->osd_plane = plane;
+                    continue;
+                }
 
-                    drm_object_free(plane);
-                    plane = NULL;
+                if (layercount == video_plane_id) {
+                    ctx->video_plane = plane;
+                    continue;
                 }
-            } else {
-                mp_err(log, "Failed to create Plane object from plane ID %d\n",
-                       drmplane->plane_id);
-                goto fail;
             }
+
+            drm_object_free(plane);
+            plane = NULL;
         }
-        drmModeFreePlane(drmplane);
-        drmplane = NULL;
     }
+    drmModeFreePlane(drmplane);
+    drmplane = NULL;
```

Comment:
Same here, maybe i didn't get the idea. but that looks a bit convoluted :)

Reply:
the drmModePlane object is not kept, however. drm_object_create only takes an id.

although I have a potentially better solution, coming up